### ISSUE TITLE
Overarching game design

### ISSUE THREAD

**Issue by digitalic** [space-wizards/space-station-14#81]
Central to the experience of creating SS14 is the idea of how we want players to engage with it.

Questions, comments, debates, pictures, comparisons between different SS13 servers,  etc. are welcome. I'm gonna try to update this issue as new points are brought up. As it gets larger we might want to bud off parts of this issue into other issues to be discussed on their own page.

---

**Core concepts for discussion:**

    Player-Player Interaction
    Player-Game Interaction
    Flow
    Player Engagement
    Game Systems

---

Programming as a game mechanic: https://github.com/space-wizards/space-station-14-content/issues/81#issuecomment-407915957, "computers" issue: https://github.com/space-wizards/space-station-14-content/issues/83

**Comment by imw-Oculus** [comment]
Core Concepts:
- Player-Player Interaction
- Player-Game Interaction
- Flow
- Player Engagement
- Game Systems

**Comment by imw-Oculus** [comment]
I still am a strong advocate of in-gaming programming. 

- We could easily turn off the ability to copy-paste, and include a wizard for making coding easier for noobs.
-  It would drastically increase the depth of the game thorugh machine modularity.
- Player-player interaction would be increased by the need to attain the required components to build a machine, and requests to build a certain machine. This is by far my weakest argument in favor.
- Programming is engaging to me, and might be to others - even if it's not the case for everyone.

**Comment by skittishZebra** [comment]
I got to say in-game programming would be engaging for me too.

Quoting @imw-Oculus , _depth is at the heart of SS13/14_, and I believe this new mechanic would make a great addition to this point. It feels too promising of a feature to not give it a try.

As for the carrying-over between rounds, isn't the same -at some point- also true for other game mechanics (I'm thinking of wikis and guides)? Perhaps the key is to design it to be potent, non-trivial & scalable.

Assembler-like programming of machine cores might just be a first step onto a whole new aspect of the game: hacking, networking, databasing, AI.

Just a thought after all, cheers.

**Comment by indeano** [comment]
Some thoughts on **story**.

The narrative experienced by consumers of SS14 is obviously fundamental to the game, so it deserves thorough examination.

Starting from the very top, we have narrative that extends the very lifespan of any server that runs SS14, from its very beginnings to its end. Think pretty much any static player character in the game. As they play more rounds, the potential for an overarching narrative emerges. When it goes right, you see a Cuban Pete, or a Bill Carson, interesting people with big impacts who change and become more defined as they are played. Notable characters, generated in the links of an interrupted chain of round-to-round playthroughs, are perpetuated outside of the game in forums, discord servers, and conversations. Is it possible to write game code that helps facilitate these cohesive, multi-round character arcs? What can be carried over between rounds? Between commits? Between transfers of server hosts?

If you count server drama, code changes, etc as "narrative" (they kind of are, a little bit), then those also carry over between rounds. Think of features expanding and taking different form as a server's population ebbs, flows, and changes culture. Can the code and design that we choose now, do something about this happening in the future?

Anyway, going deeper one level, we have the base unit of a "round" of the game. On average, they last 0.5-4 hours each, but the "persistent baystation" ss13 server actually manages to have something longer gonig for it, though (from ignorance) they probably still need to restart every so often, at the least to use new commits. There's a lot of overlapping categories of where the narrative in a round actually is, starting off with:

- The story experienced by all players during a round, versus the story experienced by all players after a round.
/
This is a worthwhile distinction. Each player might collect a "piece of the puzzle" as they play, but it's not until they get the chance to talk things out and share their experiences after the round ends that they can put things together. Things like talking to other players to find out what they did during the round, or reading others' after-action-reports posted on the server forum/imageboard. Usually text, sometimes pictures, rarely video in SS13. There's also a rudimentary way to find out what happened in a round without any involvement with its players: dive into the logs, if you even have access.
/
There's a lot that code can do here. For one, it can make "diving into the logs" a full-blown "replay the entire round for you to observe at your own pace". This feature deserves its own long post of discussion, but in short it provides massive benefits; not just to those who want to observe a round's narrative with rewinding and fast-forwarding, but to admin work, video content creators, debugging, etc. As for putting different players' narratives together, the current slate of "features" to do this is a few minutes of spammy OOC chat after roundend and talking in discord after. But what about code that generates the "highlights" of someone's round for them to share after it ends? An end-round screen that shows more than just the atnags and their objectives? What else can be done to knit together individual narratives after the round ends?

- During the round, all stories experienced with player involvement, versus all stories experienced without player involvement.
/
A weird category, but might be relevant in the future as we think about SS14's simulation becoming more complex: at some point, there will be interesting things that happen, which, by circumstance, players will not directly experience. Think of the things that you see in dwarf fortress that your dwarves don't see. Would there be a way to have the simulation heuristically detect its own interesting unobserved happenings, and bring them to the players during the game? 

- During the round, all stories that ONE player experiences, versus all stories that all other players experience.
/
This is what I consider the most important distinction to be aware of when designing a round's story. A player engaged in the game is playing as their character. The narratives the player experiences are the narratives that their character comes into contact with. At a first approximation, this is pretty much what we want; once I've invested myself into my janitor character, I care way more about my mop-fight with a clown than some security guard's batoning of the mime. But sometimes this flips around: when nothing's really going on for your guy, you'd rather find out what's happening to the nuke op being chased by the librarian. Essentially, in terms of design, we want to make sure that your character experiences the narratives that are most interesting to you at any one time.
/
What ways are there to achieve this? Generally, we want to move interesting "stories that all other players experience" into the "stories that my player experiences" category. In-character, this is already being done somewhat through in-game communications. Major interesting events that happen out of sight will tend to get yelled out over the radio, announced through consoles, or told to your character in passing. Sometimes these events can be reconstructed from the fallout. Can this pipeline be improved? How can we design the game to create in-character communications networks that only transmit interesting narratives?
/
I think there's definitely a lot that can be done. In SS13, the primary in-game other-narrative information pipeline is the chatbox. It's usually oversaturated: in any reasonably-sized server, there are too many people talking over the radio, announcements, within earshot, etc., demanding too much cognitive load for what is often too little reward, not to mention all the other crap that goes into the chatbox. Scrolling the chatbox when you miss something is also painfully bad. One easy fix is splitting off the chatbox into multiple different streams. Radio, PDA, and announcements can then be further split into their own relevant streams. Effective archiving, quick filtering, and prioritization are important: the best mark of success would be for someone latejoining an hour in to be able to find out the gist of what's happened to the station in a few minutes by simply looking things up at a console or their PDA. In general, we would aim for an adjustable balance between the player being clued into what's interesting to them, versus missing out on interesting things or being drowned in spam.

I'll add some more thoughts in a bit. Feel free to criticize, suggest, etc.

**Comment by outbools** [comment]
The answer to this is simple - ludonarrative

**Comment by digitalic** [comment]
Closing to merge into hackmd docs
